src/tecrys/svc/world/ContextBaseMusicPlayer.kt

```diff
@@ -3,50 +3,145 @@ package tecrys.svc.world
-        if(Global.getSector().playerFleet.isInHyperspace) return
-        timer.advance(amount)
-        if(timer.intervalElapsed()){
-            cooldown -= timer.intervalDuration
-            if(cooldown >= 0f) return
-            if(FleetSpawner.getFactionFleets(SVC_FACTION_ID).any {
-                it.containingLocation == Global.getSector().playerFleet.containingLocation
+        if (Global.getCurrentState() == GameState.TITLE) {
+            return
+        }
+
+        val currentTime = Global.getCombatEngine().getTotalElapsedTime(true)
+        val currentState = Global.getCurrentState()
+
+        if (lastCheckState != currentState) {
+            lastCheckTime = currentTime - 10f
+            lastCheckState = currentState
+        }
+
+        when(currentState){
+            GameState.CAMPAIGN -> {
+                stopBattleTheme()
+
+                if(Global.getSector().playerFleet.isInHyperspace) {
+                    if (isAnyVoidlingFleetInDistanceHyperspace(100f)){
+                        playExplorationTheme()
+                    } else {
+                        stopExplorationTheme()
+                    }
+                    return
+                }
+
+                if (music.isMusicPlaying(MusicID.SVC_VOIDLING_EXPLORATION_THEME) && lastCheckTime + 5f > currentTime){
+                    return
+                }
+
+                if(FleetSpawner.getFactionFleets(SVC_FACTION_ID).any {
+                    it.containingLocation == Global.getSector().playerFleet.containingLocation
                 }){
-                playSvcTheme()
-                cooldown = COOLDOWN_SECS
+                    playExplorationTheme()
+                } else if(music.isMusicPlaying(MusicID.SVC_VOIDLING_EXPLORATION_THEME)){
+                    stopExplorationTheme()
+                }
+
+                lastCheckTime = currentTime
+            }
+            GameState.COMBAT -> {
+                if (lastCheckTime + 5f > currentTime){
+                    return
+                }
+
+               if (Global.getCombatEngine()?.getFleetManager(FleetSide.ENEMY)?.deployedCopy?.filterNotNull()
+                        ?.firstOrNull()?.fleetData?.fleet?.faction == Global.getSector().getFaction(SVC_FACTION_ID)){
+                    if (!music.isMusicPlaying(MusicID.SVC_VOIDLING_BATTLE_THEME)) {
+                        playBattleTheme()
+                    }
+                } else {
+                    stopBattleTheme()
+                }
+
+                lastCheckTime = currentTime
             }
+            else -> return
         }
     }
 
-    private fun playSvcTheme(){
-        DelayedMusicPlayer.playDelayedMusic("svc_voidling_exploration_theme", 4, 1, false)
+    private fun playExplorationTheme(){
+        if (!music.isMusicPlaying(MusicID.SVC_VOIDLING_EXPLORATION_THEME)){
+            DelayedMusicPlayer.playDelayedMusic(1, 1, MusicID.SVC_VOIDLING_EXPLORATION_THEME.toMusicPlayerFormat(), false)
```

Comment:
You don't need a DelayedMusicPlayer here. It was only for triggering the music in e.g. Dialogs, since the regular sound player doesn't work there.
The way it is currently implemented, this actually causes multiple DelayedMusicPlayers to be spawned